File: dev/requirements/logic/cases/manual-provenance.5.7.case.swift
import Foundation
import SetlistCore

/// 5.7 — Every segment opened by a card tap records `manual` provenance, so
/// Phase B's proposed boundaries can never be mistaken for the comedian's own.
let manualProvenance_5_7 = RequirementCase(id: "5.7") {
    let jokes = (1 ... 3).map { Joke(title: "Bit \($0)", body: "…") }
    var stage = StageModel(jokes: jokes, plannedLength: 1200)

    stage.tapCard(jokeID: jokes[0].id, at: CaptureTime(0))
    stage.tapCard(jokeID: jokes[1].id, at: CaptureTime(120))
    stage.tapCard(jokeID: jokes[2].id, at: CaptureTime(240))
    stage.endSet(at: CaptureTime(360))

    try expectEqual(stage.segments.count, 3, "one segment per tap")
    for segment in stage.segments {
        try expectEqual(segment.provenance, .manual, "provenance of the segment at \(segment.start.seconds)s")
    }
}

File: dev/requirements/logic/cases/manual-provenance.5.7.case.dart
import 'package:setlist_core/setlist_core.dart';

import '../../shared/expectations.dart';
import '../../shared/requirement_case.dart';

/// 5.7 — Every segment opened by a card tap records `manual` provenance.
final manualProvenance_5_7 = RequirementCase('5.7', (tester) {
  final stage = StageModel(
    jokes: [
      Joke(id: 'opener', title: 'Opener', body: ''),
      Joke(id: 'van', title: "Dad's van", body: ''),
    ],
    plannedLength: const Duration(minutes: 20),
  );

  stage.tapCard(jokeId: 'opener', at: const CaptureTime(0));
  stage.tapCard(jokeId: 'van', at: const CaptureTime(60));
  stage.endSet(at: const CaptureTime(120));

  for (final segment in stage.segments) {
    expectThat(
      segment.provenance,
      SegmentProvenance.manual,
      'a boundary the comedian tapped is not one the app guessed',
    );
  }
});
